src/mixins/focus-in.js

```diff
@@ -0,0 +1,38 @@
+import { contains, eventOff, eventOn } from '../utils/dom'
+
+export default {
+  data () {
+    return {
+      focusInElement: null,
```

Comment:
The only thing about storing an element reference in data is that Vue installs observers on most of it's properties (similar to objects), although we are doing it in a few other components as we need to see if the element is changed.

Reply:
What about using a ref name instead and resolve `document` when no ref name is defined?

This isn't particularly needed for the current use case but makes this mixin much more flexible for a usage in other parts of the project.

Reply:
That might be an option.. or make it a non reactive private property.. unless it needs to be reactive 5o changes in the element ref.

Reply:
To make it non reactive, just don't define it in the data.